ast/functions_test.go

```diff
@@ -227,3 +227,17 @@ func (ts *testDMLSuite) TestWindowFuncExprRestore(c *C) {
 	}
 	RunNodeRestoreTest(c, testCases, "select %s from t", extractNodeFunc)
 }
+
+func (ts *testFunctionsSuite) TestGenericFuncRestore(c *C) {
+	testCases := []NodeRestoreTestCase{
+		{"s.a()", "`s`.A()"},
+		{"`s`.`a`()", "`s`.A()"},
+		{"now()", "NOW()"},
+		{"`s`.`now`()", "`s`.NOW()"},
+		{"generic_func()", "GENERIC_FUNC()"},
```

Comment:
The restored SQL text is very strange because the function name is written as keyword:
```
ctx.WriteKeyWord(n.FnName.O)
```

- If we use `WriteName()`, it restores `now()` to `` `now`() ``, which is invalid in MySQL:
```
FUNCTION test.now does not exist. Check the 'Function Name Parsing and Resolution' section in the Reference Manual
```

Reply:
Perhaps we need one more field to indicate whether the function is a name or keyword (according to rules in https://dev.mysql.com/doc/refman/8.0/en/function-resolution.html)